crates/bevy_egui/src/render.rs

```diff
@@ -0,0 +1,240 @@
+                ..Default::default()
+            },
+            visible_entities: Default::default(),
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, far - 0.1)),
+            global_transform: Default::default(),
+        }
+    }
+}
+
+pub const EGUI_PIPELINE_HANDLE: Handle<PipelineDescriptor> =
+    // No clue where the hell this number comes from, but I changed a few of them to be slightly different
+    Handle::from_u128(323432602125399287835112542539754486265);
+
+pub fn build_egui_pipeline(shaders: &mut Assets<Shader>) -> PipelineDescriptor {
+    PipelineDescriptor {
+        index_format: IndexFormat::Uint16,
+        primitive_topology: PrimitiveTopology::TriangleList,
+
+        rasterization_state: Some(RasterizationStateDescriptor {
+            front_face: FrontFace::Ccw,
+            cull_mode: CullMode::None,
+            depth_bias: 0,
+            depth_bias_slope_scale: 0.0,
+            depth_bias_clamp: 0.0,
+            clamp_depth: false,
+        }),
+        depth_stencil_state: Some(DepthStencilStateDescriptor {
+            format: TextureFormat::Depth32Float,
+            depth_write_enabled: false,
+            depth_compare: CompareFunction::Always,
+            stencil: StencilStateDescriptor {
+                front: StencilStateFaceDescriptor::IGNORE,
+                back: StencilStateFaceDescriptor::IGNORE,
+                read_mask: 0,
+                write_mask: 0,
+            },
+        }),
+        color_states: vec![ColorStateDescriptor {
+            format: TextureFormat::Bgra8UnormSrgb,
+            color_blend: BlendDescriptor {
+                src_factor: BlendFactor::SrcAlpha,
+                dst_factor: BlendFactor::OneMinusSrcAlpha,
+                operation: BlendOperation::Add,
+            },
+            alpha_blend: BlendDescriptor {
+                src_factor: BlendFactor::OneMinusDstAlpha,
+                dst_factor: BlendFactor::One,
+                // src_factor: BlendFactor::One,
+                // dst_factor: BlendFactor::OneMinusSrcAlpha,
+                operation: BlendOperation::Add,
+            },
+            write_mask: ColorWrite::ALL,
+        }],
+        ..PipelineDescriptor::new(ShaderStages {
+            vertex: shaders.add(Shader::from_glsl(
+                ShaderStage::Vertex,
+                include_str!("egui.vert"),
+            )),
+            fragment: Some(shaders.add(Shader::from_glsl(
+                ShaderStage::Fragment,
+                include_str!("egui.frag"),
+            ))),
+        })
+    }
+}
+
+pub mod node {
+    pub const EGUI_CAMERA_ID: &str = "egui_camera";
+    pub const EGUI_SYSTEM_NODE_ID: &str = "egui_system_node";
+    pub const EGUI_NODE_ID: &str = "egui_node";
+    pub const EGUI_PASS_ID: &str = "egui_pass";
+}
+
+pub mod camera {
+    pub const EGUI_CAMERA_NAME: &str = "EguiCamera";
+}
+
+pub trait EguiRenderGraphBuilder {
+    fn add_egui_graph(&mut self, resources: &Resources) -> &mut Self;
+}
```

Comment:
Add comments describing the purpose of the trait, as well as the intended purpose and actions of `add_egui_graph`.